### PR TITLE
Added tracing API

### PR DESCRIPTION
**Pull request by MadelineMurray**
Added tracing API

### CODE REVIEW

File: docs/Concepts/Transactions/Trace-Types.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+For out of gas operations, Besu reports the operation that caused the out of gas exception including
+the calculated gas cost. No `ex` values are reported because the operation is not executed. 
+
+## stateDiff
+
+State changes in the requested block for each transaction represented as 
+a map of accounts to an object. The balance, code, nonce, and storage changes are enumerated
+from immediately before the transaction to after the transaction.  For the `key:value` pairs:  
+
+* `+` indicates the field didn’t exist before and how has the specified value
```
Comment (bgravenorst) [inline docs/Concepts/Transactions/Trace-Types.md]: typo "how" -> "now"?

File: docs/Concepts/Transactions/Trace-Types.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    },
+    "storage":{              
+    }
+  }
+}
+``` 
+
+| Key            | Value                                                                                |
+|-----------     | --------------------------------------------------------------------------------------|
+| `balance`      | Change of balance event
```
Comment (bgravenorst) [inline docs/Concepts/Transactions/Trace-Types.md]: The descriptions in the previous table ended with fullstops (.).

File: docs/HowTo/Troubleshoot/Trace-Transactions.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# Trace Transactions 
+
+Use the `TRACE` API to get detailed information about transaction processing. Enable the `TRACE` API using 
+the [`-rpc-http-api`](../../Reference/CLI/CLI-Syntax.md#rpc-http-api) or [`rpc-ws-api`](../../Reference/CLI/CLI-Syntax.md#rpc-ws-api) 
+command line options. 
+
+Use the [`trace_replayBlockTransactions`](../../Reference/API-Methods.md#trace_replayblocktransactions) 
+JSON RPC API method and specify the trace types required. Options are [`trace`, `vmTrace`, or `stateDiff`](../../Concepts/Transactions/Trace-Types.md). 
+
+Your node must be either an archive node (that is, synchronised without using pruning or fast sync)
```
Comment (bgravenorst) [inline docs/HowTo/Troubleshoot/Trace-Transactions.md]: Can probably remove "either"

File: docs/Reference/API-Objects.md [hunk 4]
```diff
@@ -189,9 +189,22 @@ Returned by [eth_getTransactionReceipt](API-Methods.md#eth_gettransactionreceipt
...
 | **root** | Data, 32&nbsp;bytes| Post-transaction stateroot|
 
+## Transaction Trace Object 
+
+Returned by [`trace_replayBlockTransactions`](API-Methods.md#trace_replayblocktransactions). 
+
+| Key                                  | Type                         | Value                                                                                |
+|-----------------------               |:--------------------:        |--------------------------------------------------------------------------------------|
+| **output**                           | ?                            | Transaction result 
+| **stateDiff**                        | Object                       | [State changes in the requested block](../Concepts/Transactions/Trace-Types.md)
```
Comment (bgravenorst) [inline docs/Reference/API-Objects.md]: Minor. You may want to link to the exact heading in the topic.

File: docs/Concepts/Transactions/Trace-Types.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+# Transaction Trace Types  
+
+When using [`trace_replayBlockTransactions`](../../Reference/API-Methods.md#trace_replayblocktransactions)
+the trace options are `trace`, `vmTrace`, and `stateDiff`.  
+
+## trace  
+
+Ordered list of calls to other contracts excluding precompiled contracts.  Indicates whether the 
+transaction is a value transfer or contract call, gas provided, gas used, from and to addresses,
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Concepts/Transactions/Trace-Types.md]: Done

File: docs/Concepts/Transactions/Trace-Types.md (hunk 1 above)
Comment (MadelineMurray) [reply docs/Concepts/Transactions/Trace-Types.md]: Fixed

File: docs/Concepts/Transactions/Trace-Types.md (hunk 2 above)
Comment (MadelineMurray) [reply docs/Concepts/Transactions/Trace-Types.md]: Added

File: docs/HowTo/Troubleshoot/Trace-Transactions.md (hunk 3 above)
Comment (MadelineMurray) [reply docs/HowTo/Troubleshoot/Trace-Transactions.md]: Done

File: docs/Reference/API-Objects.md (hunk 4 above)
Comment (MadelineMurray) [reply docs/Reference/API-Objects.md]: Done

**Pull request closed without merging**